### ISSUE TITLE
Fix the notification issue when the author's displayname needs to be subject parameter

### ISSUE THREAD

**Issue by gig13** [owncloud/announcementcenter#58]
Announcement center 1.0.2 for 8.2 puts the uid of the author of the announcement in the notification. So end users see this uid instead of the author's name when they click their little bell icon. Within the announcement center app itself the display name is used except immediately after an announcement is written. When the announcement is saved and appears in the list at the bottom of the form it shows the author's uid. Reload the page and display name appears. Probably an ajax issue. Admins only see it so it's not a real problem.

To fix the notification issue the author's displayname needs to be set as the subject parameter instead of the uid.

```
announcementcenter/controller/pagecontroller.php line 185 to 189:
$notification->setApp('announcementcenter')
->setTimestamp($timeStamp)
->setObject('announcement', $id)
->setSubject('announced', [$this->userManager->get($authorId)->getDisplayName()])
->setLink($this->urlGenerator->linkToRoute('announcementcenter.page.index'));
```

@MorrisJobke A friendly update thanks to @scolebrook

**Comment by nickvergessen** [comment]
Will have a look, thanks

**Comment by nickvergessen** [comment]
The fix is on the wrong possition, we should select the display name on displaying the notification, not on inserting. Otherwise changes are not reflected.

But I will fix this for 9.0.1
